bindings.go

```diff
@@ -730,6 +736,215 @@ func GeneratePieceCommitmentFromFile(pieceFile *os.File, pieceSize uint64) (comm
+	cacheDirPath string,
+	stagedSectorPath string,
+	sealedSectorPath string,
+	sectorID uint64,
+	proverID [CommitmentBytesLen]byte,
+	ticket [CommitmentBytesLen]byte,
+	pieces []PublicPieceInfo,
+) (RawSealPreCommitOutput, error) {
+	defer elapsed("StandaloneSealPreCommit")()
+
+	cCacheDirPath := C.CString(cacheDirPath)
+	defer C.free(unsafe.Pointer(cCacheDirPath))
+
+	cStagedSectorPath := C.CString(stagedSectorPath)
+	defer C.free(unsafe.Pointer(cStagedSectorPath))
+
+	cSealedSectorPath := C.CString(sealedSectorPath)
+	defer C.free(unsafe.Pointer(cSealedSectorPath))
+
+	proverIDCBytes := C.CBytes(proverID[:])
+	defer C.free(proverIDCBytes)
+
+	ticketCBytes := C.CBytes(ticket[:])
+	defer C.free(ticketCBytes)
+
+	cPiecesPtr, cPiecesLen := cPublicPieceInfo(pieces)
+	defer C.free(unsafe.Pointer(cPiecesPtr))
+
+	resPtr := C.sector_builder_ffi_reexported_seal_pre_commit(
+		cSectorClass(sectorSize, poRepProofPartitions),
+		cCacheDirPath,
+		cStagedSectorPath,
+		cSealedSectorPath,
+		C.uint64_t(sectorID),
+		(*[32]C.uint8_t)(proverIDCBytes),
+		(*[32]C.uint8_t)(ticketCBytes),
+		(*C.sector_builder_ffi_FFIPublicPieceInfo)(cPiecesPtr),
+		cPiecesLen,
+	)
+	defer C.sector_builder_ffi_reexported_destroy_seal_pre_commit_response(resPtr)
+
+	if resPtr.status_code != 0 {
+		return RawSealPreCommitOutput{}, errors.New(C.GoString(resPtr.error_msg))
+	}
+
+	return goRawSealPreCommitOutput(resPtr.seal_pre_commit_output), nil
+}
+
+// StandaloneSealCommit
+func StandaloneSealCommit(
+	sectorSize uint64,
+	poRepProofPartitions uint8,
+	cacheDirPath string,
+	sectorID uint64,
+	proverID [CommitmentBytesLen]byte,
+	ticket [CommitmentBytesLen]byte,
+	seed [CommitmentBytesLen]byte,
+	pieces []PublicPieceInfo,
+	rspco RawSealPreCommitOutput,
+) ([]byte, error) {
+	defer elapsed("StandaloneSealCommit")()
+
+	cCacheDirPath := C.CString(cacheDirPath)
+	defer C.free(unsafe.Pointer(cCacheDirPath))
+
+	proverIDCBytes := C.CBytes(proverID[:])
+	defer C.free(proverIDCBytes)
+
+	ticketCBytes := C.CBytes(ticket[:])
+	defer C.free(ticketCBytes)
+
+	seedCBytes := C.CBytes(seed[:])
+	defer C.free(seedCBytes)
+
+	cPiecesPtr, cPiecesLen := cPublicPieceInfo(pieces)
+	defer C.free(unsafe.Pointer(cPiecesPtr))
+
+	resPtr := C.sector_builder_ffi_reexported_seal_commit(
+		cSectorClass(sectorSize, poRepProofPartitions),
+		cCacheDirPath,
+		C.uint64_t(sectorID),
+		(*[32]C.uint8_t)(proverIDCBytes),
+		(*[32]C.uint8_t)(ticketCBytes),
+		(*[32]C.uint8_t)(seedCBytes),
+		(*C.sector_builder_ffi_FFIPublicPieceInfo)(cPiecesPtr),
+		cPiecesLen,
+		cSealPreCommitOutput(rspco),
+	)
+	defer C.sector_builder_ffi_reexported_destroy_seal_commit_response(resPtr)
+
+	if resPtr.status_code != 0 {
+		return nil, errors.New(C.GoString(resPtr.error_msg))
+	}
+
+	return C.GoBytes(unsafe.Pointer(resPtr.proof_ptr), C.int(resPtr.proof_len)), nil
+}
+
+// StandaloneUnseal
+func StandaloneUnseal(
+	sectorSize uint64,
+	poRepProofPartitions uint8,
+	sealedSectorPath string,
+	unsealOutputPath string,
```

Comment:
It would be really nice if this was a writable `os.File` (Ignore if unseal isn't sequential)

Also, is there any way this could support offset / size? (not necesarily now, just theoretically)

Reply:
@magik6k 

> It would be really nice if this was a writable os.File (Ignore if unseal isn't sequential)

I agree. It would be nice. Unfortunately, the rust-fil-proofs [unsealing function](https://github.com/filecoin-project/rust-fil-proofs/blob/master/filecoin-proofs/src/api/mod.rs#L40) requires [that its output location](https://github.com/filecoin-project/rust-fil-proofs/blob/master/filecoin-proofs/src/fr32.rs#L650) (in this case, a file descriptor to `unsealOutputPath` is read/write/fseek-able.

> Also, is there any way this could support offset / size? (not necesarily now, just theoretically)

Yes. The [rust-fil-proofs `get_unsealed_range` function](https://github.com/filecoin-project/rust-fil-proofs/blob/master/filecoin-proofs/src/api/mod.rs#L40) allows a caller to partially unseal. I didn't expose that API as part of this PR, but can do so in the future. Note that you must decode the full sector - even if you're interested in a subset of its data. So, unsure how useful that API would be.

Reply:
> I agree. It would be nice. Unfortunately, the rust-fil-proofs unsealing function requires that its output location (in this case, a file descriptor to unsealOutputPath which is read/write/fseek-able).

That's fine for now, but will be nice to have in the future

> Yes. The rust-fil-proofs get_unsealed_range function allows a caller to partially unseal. I didn't expose that API as part of this PR, but can do so in the future.

Yes, please. (Allows sectorbuilder to do some nice caching)